### DISCUSSION TITLE
Migration from Airbnb's ESLint to Next's ESLint

### DISCUSSION THREAD

**Discussion by bnhr** [vercel/next.js#26577] [Help] [answered]
I'm using Airbnb's ESLinnt config for my Next v.10.2 projects and want to upgrade to Next v.11.
Is there any specific migration about ESLint? or I can just simply remove everything related to Airbnb and start type `yarn lint`?

**Comment by housseindjirdeh** [comment, marked as answer]
There's two options we recommend you choosing between:

1. If you're not particularly tied to using Airbnb's configuration and want `react`, `react-hooks`, `import`, and `next` plugin support - remove `eslint-config-airbnb` and switch to using `eslint-config-next`. You can do this by just removing your entire `.eslintrc` file and running `yarn lint`
2. If you still want to use `eslint-config-airbnb` for any particular reason, install `eslint-plugin-next` instead and extend from their to get additional Next.js-specific rules ([details](https://nextjs.org/docs/basic-features/eslint#migrating-existing-config)).

**Reply by Duncan-Brain** [reply]
Hi @housseindjirdeh ,

I have been stuck deciding between the airbnb and next configs for the past day or so. It is my first week of linting and I was hoping you might be able to elaborate a bit more about option 2.  I find myself wanting some of the stricter style enforcements that the airbnb-config has chosen... but I also don't want to miss out on important gotchas in how I interact with my Next framework. 

Are there any supplementary suggestions for the eslint-plugin-next in terms of rules, settings, etc? 

For example I notice the eslint-config-next has rules like 
```
 'jsx-a11y/alt-text': [
      'warn',
      {
        elements: ['img'],
        img: ['Image'],
      },
    ],
```
This seems like an useful and important rule to add to my configuration -- and it is also very Next.js specific. It has got me thinking that there are possibly other settings available in the eslint-config-next setting that I don't immediately understand the usefulness of but could be important for optimal linting. This example above with jsx-a11y/alt-text could possibly be useful to add to the eslint-plugin-next documentation --- despite not technically extending the Next.js specific rules. 

It also reminded me of one of the suggested configurations in the [eslint-plugin-react ReadMe](https://github.com/yannickcr/eslint-plugin-react/) 
   ```
 "linkComponents": [
      // Components used as alternatives to <a> for linking, eg. <Link to={ url } />
      "Hyperlink",
      {"name": "Link", "linkAttribute": "to"}
    ]
```
I could not find this react plugin configuration in the eslint-next-config and thought it would certainly apply for next/link. But there could be something I am missing.

Overall, about the eslint-config-next I was curious: Which are Next.js team preferential rules and settings? And which are really Next.js specific to avoid gotchas? Lastly do you have any other resources about why some of the suggested react settings don't need to be applied?

Thank you so much for your time,
Duncan
